File: test_shopee_last_try/5a.r
library(dplyr)
library(tidyr)

order <- read.csv("Analisis excel/Youtube/5 soal/Orders.csv")
order$Order.Date <- as.Date(order$Order.Date, "%m/%d/%Y")
tanggal <- order %>%
    group_by(Customer.ID) %>%
    summarise(first_order = min(Order.Date))

df <- inner_join(order, tanggal, by = "Customer.ID") %>% filter(Order.Date == first_order)
print(head(df))

hasil <- subset(df, select = c(Customer.ID, Sales)) %>% arrange(Customer.ID)
print(head(hasil))
print(dim(hasil))

cat("\n")
hasil2 <- df %>%
    group_by(Customer.ID) %>%
    summarise(total = list(unique(Sales))) %>%
    unnest(total) %>%
    arrange(Customer.ID)
print(head(hasil2))
print(dim(hasil2))

File: test_shopee_last_try/5a.py
import pandas as pd

df = pd.read_csv('Analisis excel/test_shopee_last_try/Orders.csv')
df['Order Date'] = pd.to_datetime(df['Order Date'])
filter = df.groupby('Customer ID')['Order Date'].min().reset_index(name='tanggal_pertama')
hasil_filter = pd.merge(df, filter, on='Customer ID', how='inner')
hasil_filter = hasil_filter[hasil_filter['Order Date'] == hasil_filter['tanggal_pertama']].sort_values('Order Date', ascending=False).reset_index(drop=True)
# print(hasil_filter)

hasil = hasil_filter[['Customer ID', 'Sales']].sort_values('Customer ID', ascending=True).reset_index(drop=True)
print(hasil)

hasil = hasil_filter.groupby('Customer ID')['Sales'].unique().explode().reset_index().sort_values('Customer ID', ascending=True).reset_index(drop=True)
print(hasil)
# hasil2 = hasil_filter['Order ID'].value_counts()
# print(hasil2)